src/postgres/src/backend/utils/adt/dbsize.c

```diff
@@ -481,14 +481,51 @@ calculate_indexes_size(Relation rel)
 		{
 			Oid			idxOid = lfirst_oid(cell);
 			Relation	idxRel;
-			ForkNumber	forkNum;
 
 			idxRel = relation_open(idxOid, AccessShareLock);
 
-			for (forkNum = 0; forkNum <= MAX_FORKNUM; forkNum++)
-				size += calculate_relation_size(&(idxRel->rd_node),
-												idxRel->rd_backend,
-												forkNum);
+			if (IsYBRelation(idxRel))
+			{
+				/* Primary index relation doesn't have dedicated table in DocDB */
+				if (idxRel->rd_index && idxRel->rd_index->indisprimary)
+				{
+					relation_close(idxRel, AccessShareLock);
+					continue;
+				}
+
+				/* Colcoated tables do not have size info */
+				if (YbGetTableProperties(idxRel)->is_colocated)
+				{
+					relation_close(idxRel, AccessShareLock);
+					continue;
+				}
+
+				int64		index_size = 0;
+				int32		num_missing_tablets = 0;
+
+				HandleYBStatus(YBCPgGetTableDiskSize(YbGetRelfileNodeId(idxRel),
+													 YBCGetDatabaseOid(idxRel), (int64_t *) &index_size, &num_missing_tablets));
```

Comment:
Do we really need a typecast here?
Can we declare the local variable as `int64_t` to avoid it?